cas_implementations/src/eav/file.rs

```diff
@@ -94,88 +122,131 @@ impl EavFileStorage {
         &self,
         subscript: String,
         eav_constraint: Option<T>,
-    ) -> HashSet<HcResult<String>>
+    ) -> HcResult<BTreeSet<String>>
     where
         T: ToString,
     {
         let address = eav_constraint
             .map(|e| e.to_string())
-            .unwrap_or(String::new());
-        let full_path =
-            vec![self.dir_path.clone(), subscript, address].join(&MAIN_SEPARATOR.to_string());
-        let mut set = HashSet::new();
-        WalkDir::new(full_path.clone())
-            .into_iter()
-            .for_each(|dir_entry| match dir_entry {
-                Ok(eav_content) => {
-                    add_eav_to_hashset(eav_content, &mut set);
-                }
-                Err(_) => {
-                    set.insert(Err(HolochainError::IoError(format!(
-                        "Could not obtain directory{:?}",
-                        full_path
-                    ))));
-                }
-            });
-
-        set
+            .unwrap_or(String::from("*"));
+        let path = vec![self.dir_path.clone(), subscript].join(&MAIN_SEPARATOR.to_string());
+        if Path::new(&path.clone()).exists() {
+            let full_path = vec![path.clone(), address.clone()].join(&MAIN_SEPARATOR.to_string());
+
+            let paths = glob(&*full_path.clone())
+                .map_err(|_| HolochainError::ErrorGeneric("Could not get form path".to_string()))?;
+
+            let (eavs, errors): (Vec<_>, Vec<_>) = paths
+                .map(|path_val| {
+                    path_val.map(|walk| read_eav(walk.clone())).unwrap_or(Err(
+                        HolochainError::ErrorGeneric(
+                            "Could not read eavs from directory".to_string(),
+                        ),
+                    ))
+                })
+                .partition(Result::is_ok);
+            if errors.len() > 0 {
+                Err(HolochainError::ErrorGeneric(
+                    "Could not read eavs from directory".to_string(),
+                ))
+            } else {
+                let mut ordmap: BTreeSet<String> = BTreeSet::new();
+                eavs.iter().for_each(|s| {
+                    s.clone().unwrap_or(Vec::new()).iter().for_each(|value| {
+                        ordmap.insert(value.clone());
+                    })
+                });
+                Ok(ordmap)
+            }
+        } else {
+            Ok(BTreeSet::new())
+        }
     }
 }
 
 impl EntityAttributeValueStorage for EavFileStorage {
-    fn add_eav(&mut self, eav: &EntityAttributeValue) -> Result<(), HolochainError> {
+    fn add_eav(
+        &mut self,
+        eav: &EntityAttributeValueIndex,
```

Comment:
eavi?

Reply:
yes those will need to be changed to that